packages/framework/src/Form/Constraints/FileAbstractFilesystemValidator.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Form\Constraints;
+
+use League\Flysystem\MountManager;
+use Shopsys\FrameworkBundle\Component\FileUpload\FileUpload;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
+use Symfony\Component\HttpFoundation\File\File as FileObject;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Constraints\FileValidator;
+
+class FileAbstractFilesystemValidator extends FileValidator
+{
+    /**
+     * @var \League\Flysystem\MountManager
+     */
+    private $mountManager;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\FileUpload\FileUpload
+     */
+    private $fileUpload;
+
+    /**
+     * @var \Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface
+     */
+    private $parameterBag;
+
+    /**
+     * @param \League\Flysystem\MountManager $mountManager
+     * @param \Shopsys\FrameworkBundle\Component\FileUpload\FileUpload $fileUpload
+     * @param \Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface $parameterBag
+     */
+    public function __construct(MountManager $mountManager, FileUpload $fileUpload, ParameterBagInterface $parameterBag)
+    {
+        $this->mountManager = $mountManager;
+        $this->fileUpload = $fileUpload;
+        $this->parameterBag = $parameterBag;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        $abstractPath = $this->fileUpload->getTemporaryFilepath($value->getFilename());
+        $localPath = $this->parameterBag->get('shopsys.root_dir') . $this->fileUpload->getTemporaryFilepath(uniqid() . $value->getFilename());
```

Comment:
142 characters on one line are not really good for reading. Could you make it more readable?

Reply:
fixed